topo/node/ceos/ceos.go

```diff
@@ -66,42 +79,95 @@ func New(nodeImpl *node.Impl) (node.Node, error) {
 	n := &Node{
 		Impl: nodeImpl,
 	}
-	n.FixInterfaces()
+	err := n.FixInterfaces()
+	if err != nil {
+		return nil, err
+	}
 	return n, nil
 }
 
-// SpawnCLIConn spawns a CLI connection towards a Network OS using `kubectl exec` terminal and ensures CLI is ready
-// to accept inputs.
-// scrapligo options can be provided to this function for a caller to modify scrapligo platform.
-// For example, mock transport can be set via options
-func (n *Node) SpawnCLIConn() error {
-	opts := []scrapliutil.Option{
-		scrapliopts.WithAuthBypass(),
+func (n *Node) Create(ctx context.Context) error {
+	if err := n.CreateConfig(ctx); err != nil {
+		return fmt.Errorf("node %s failed to create config-map %w", n.Name(), err)
 	}
-
-	// add options defined in test package
-	opts = append(opts, n.testOpts...)
-
-	opts = n.PatchCLIConnOpen("kubectl", []string{"Cli"}, opts)
-
-	var err error
-	n.cliConn, err = n.GetCLIConn(scrapliPlatformName, opts)
-
-	return err
+	if err := n.CreateCRD(ctx); err != nil {
+		return fmt.Errorf("node %s failed to create custom resource definition %w", n.Name(), err)
+	}
+	return nil
 }
 
-func (n *Node) GenerateSelfSigned(ctx context.Context) error {
-	selfSigned := n.Proto.GetConfig().GetCert().GetSelfSigned()
-	if selfSigned == nil {
-		log.Infof("%s - no cert config", n.Name())
-		return nil
+func (n *Node) CreateCRD(ctx context.Context) error {
+	log.Infof("Creating new CEosLabDevice CRD for node: %v", n.Name())
+	proto := n.GetProto()
+	config := proto.GetConfig()
+	device := &ceos.CEosLabDevice{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "ceoslab.arista.com/v1alpha1",
+			Kind:       "CEosLabDevice",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      n.Name(),
+			Namespace: n.GetNamespace(),
+			Labels: map[string]string{
+				"app":  n.Name(),
+				"topo": n.GetNamespace(),
+			},
+		},
+		Spec: ceos.CEosLabDeviceSpec{
+			EnvVar:        config.GetEnv(),
+			Image:         config.GetImage(),
```

Comment:
we also need the init image field to be configurable

Reply:
Pushed arista-ceoslab-operator v0.2.0 and updated PR